crates/orchestrator/src/tests/jobs/state_update_job/mod.rs

```diff
@@ -21,16 +22,98 @@ use crate::jobs::{
 use crate::config::{config, config_force_init};
 use crate::constants::{BLOB_DATA_FILE_NAME, SNOS_OUTPUT_FILE_NAME};
 use crate::data_storage::MockDataStorage;
-use crate::jobs::state_update_job::utils::hex_string_to_u8_vec;
+use crate::jobs::constants::JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO;
+use crate::jobs::state_update_job::utils::{fetch_blob_data_for_block, hex_string_to_u8_vec};
+use crate::tests::common::{default_job_item, get_storage_client};
+use crate::tests::config::TestConfigBuilder;
 use httpmock::prelude::*;
 use lazy_static::lazy_static;
+use utils::env_utils::get_env_var_or_panic;
 
 lazy_static! {
     pub static ref CURRENT_PATH: PathBuf = std::env::current_dir().unwrap();
 }
 
 pub const X_0_FILE_NAME: &str = "x_0.txt";
 
+// ================= Exhaustive tests (with minimum mock) =================
+
+#[rstest]
+#[should_panic(expected = "Could not find current attempt number.")]
+#[tokio::test]
+async fn test_process_job_attempt_not_present_fails() {
+    TestConfigBuilder::new().build().await;
+
+    let mut job = default_job_item();
+    let config = config().await;
+    let state_update_job = StateUpdateJob {};
+    let _ = state_update_job.process_job(&config, &mut job).await;
+}
+
+#[rstest]
+#[case(None, String::from("651053,651054,651055"), 0)]
+#[case(Some(651054), String::from("651053,651054,651055"), 1)]
+#[tokio::test]
+async fn test_process_job_works(
+    #[case] failed_block_number: Option<u64>,
+    #[case] blocks_to_process: String,
+    #[case] processing_start_index: u8,
+) {
+    // Will be used by storage client which we call while storing the data.
+    dotenvy::from_filename("../.env.test").expect("Failed to load the .env file");
+
+    // Mocking the settlement client.
+    let mut settlement_client = MockSettlementClient::new();
+
+    let block_numbers: Vec<u64> = parse_block_numbers(&blocks_to_process).unwrap();
+
+    // This must be the last block number and should be returned as an output from the process job.
+    let last_block_number = block_numbers[block_numbers.len() - 1];
+
+    // Storing `blob_data` and `snos_output` in storage client
+    store_data_in_storage_client_for_s3(block_numbers.clone()).await;
+
+    // Building a temp config that will be used by `fetch_blob_data_for_block` and `fetch_snos_for_block`
+    // functions while fetching the blob data from storage client.
+    TestConfigBuilder::new().build().await;
+
+    // Adding expectations for each block number to be called by settlement client.
+    for (i, _) in block_numbers.iter().enumerate().skip(processing_start_index as usize) {
```

Comment:
```suggestion
    for i in (processing_start_index as usize)..block_numbers.len() {
```

Reply:
That sounds weird. Did you try this exact syntax?

Reply:
yeah little bit different than this. but sure I will try your syntax and update it here.

Reply:
For you syntax also this is the error I am getting, which is same when I used this syntax. 

```sh
error: the loop variable `i` is only used to index `block_numbers`
  --> crates/orchestrator/src/tests/jobs/state_update_job/mod.rs:79:14
   |
79 |     for i in (processing_start_index as usize)..block_numbers.len() {
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
   = note: `-D clippy::needless-range-loop` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::needless_range_loop)]`
help: consider using an iterator
   |
79 |     for <item> in block_numbers.iter().skip((processing_start_index as usize)) {
   |         ~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Reply:
Ah yeah but then Clippy is giving you the right advice, you do not need `i` at all. Just apply the suggestion of Clippy.